Automattic/pocket-casts-android
# Add gitleaks pre commit hook

## Description

Adds a gitleaks check to the pre-commit hook. Having this makes me feel much more comfortable that we're not going to accidentally leak a secret. It is adding another pre-commit hook though, which there are good reasons to avoid if not necessary, so let me know if you think this isn't a good idea.

I didn't love the adding this to the commit hook for any open-source contributors though since it requires installing gitleaks, so I'm skipping the gitleaks part of the hook if the user does not have a `sentry.properties` file. I don't feel strongly about this specific check, it was just something fresh in my mind that I knew was necessary for making a release build. I definitely open to other suggestions.

## Test steps

1. Run `./gradlew installGitHooks`
2. Ensure you have a `sentry.properties` file because I'm using that to detect whether or not to run gitleaks.
3. Try to commit something that gitleaks would detect as a secret. For example, the file from running this command: `echo "https://my-private-feed.com/rss/618?auth=snZc3xsAd3GCMVvY6ZSfymsQ" > file.txt`
4. Observe that the commit is rejected.
5. Delete or rename `sentry.properties` file
6. Try to commit the file with the fake secret again
7. Observe that the commit succeeds.

# Checklist
